src/app/core/components/gift-storage/gift-storage.component.spec.ts

```diff
@@ -0,0 +1,82 @@
+import { AccountService } from '@shared/services/account/account.service';
+/* @format */
+import { Shallow } from 'shallow-render';
+import { HttpClient, HttpHandler } from '@angular/common/http';
+import { CoreModule } from '@core/core.module';
+import { Dialog } from '@root/app/dialog/dialog.service';
+import { AccountVO } from '../../../models/account-vo';
+import { GiftStorageComponent } from './gift-storage.component';
+
+describe('GiftStorageComponent', () => {
+  let shallow: Shallow<GiftStorageComponent>;
+
+  const mockAccount = new AccountVO({ accountId: 1 });
+
+  const mockAccountService = {
+    getAccount: jasmine.createSpy('getAccount').and.returnValue({
+      mockAccount,
+    }),
+  };
+
+  beforeEach(() => {
+    shallow = new Shallow(GiftStorageComponent, CoreModule)
+      .provide([HttpClient, HttpHandler])
+      .mock(AccountService, mockAccountService)
+      .mock(Dialog, { open: (token, data, options) => Promise.resolve({}) });
+  });
+
+  it('should create', async () => {
+    const { instance } = await shallow.render(); // Render the component
+    expect(instance).toBeTruthy();
+  });
+
+  it('enables the "Send Gift Storage" button when the form is valid', async () => {
+    const { find, instance, fixture } = await shallow.render();
+
+    instance.availableSpace = '10';
+
+    instance.giftForm.controls.email.setValue('test@example.com');
+    instance.giftForm.controls.amount.setValue('1');
+
+    instance.giftForm.updateValueAndValidity();
+    fixture.detectChanges();
+
+    const button: HTMLButtonElement = find('.btn-primary').nativeElement;
+
+    expect(button.disabled).toBe(false);
+  });
+
+  it('the button is disabled if the email is not valid', async () => {
```

Comment:
Nitpicky, but it seems like the convention for test names is to use a more active sentence, like "disables the button if the email is not valid."